Sources/OpenAPIRuntime/Base/Acceptable.swift

```diff
@@ -0,0 +1,167 @@
+
+    /// Creates a new quality value of the default value 1.0.
+    public init() {
+        self.thousands = 1000
+    }
+
+    /// Returns a Boolean value indicating whether the quality value is
+    /// at its default value 1.0.
+    public var isDefault: Bool {
+        thousands == 1000
+    }
+
+    /// Creates a new quality value from the provided floating-point number.
+    ///
+    /// - Precondition: The value must be between 0.0 and 1.0, inclusive.
+    public init(doubleValue: Double) {
+        precondition(
+            doubleValue >= 0.0 && doubleValue <= 1.0,
+            "Provided quality number is out of range, must be between 0.0 and 1.0, inclusive."
+        )
+        self.thousands = UInt16(doubleValue * 1000)
+    }
+
+    /// The value represented as a floating-point number between 0.0 and 1.0, inclusive.
+    public var doubleValue: Double {
+        Double(thousands) / 1000
+    }
+}
+
+extension QualityValue: RawRepresentable {
+    public init?(rawValue: String) {
+        guard let doubleValue = Double(rawValue) else {
+            return nil
+        }
+        self.init(doubleValue: doubleValue)
+    }
+
+    public var rawValue: String {
+        String(format: "%0.3f", doubleValue)
+    }
+}
+
+extension QualityValue: ExpressibleByIntegerLiteral {
+    public init(integerLiteral value: UInt16) {
+        precondition(
+            value >= 0 && value <= 1,
+            "Provided quality number is out of range, must be between 0 and 1, inclusive."
+        )
+        self.thousands = value * 1000
+    }
+}
+
+extension QualityValue: ExpressibleByFloatLiteral {
+    public init(floatLiteral value: Double) {
+        self.init(doubleValue: value)
+    }
+}
+
+extension Array {
+
+    /// Returns the default values for the acceptable type.
+    public static func defaultValues<T: AcceptableProtocol>() -> [AcceptHeaderContentType<T>]
+    where Element == AcceptHeaderContentType<T> {
+        T.allCases.map { .init(contentType: $0) }
+    }
+}
+
+/// A wrapper of an individual content type in the accept header.
+public struct AcceptHeaderContentType<ContentType: AcceptableProtocol>: Sendable, Hashable {
+
+    /// The value representing the content type.
+    public var contentType: ContentType
+
+    /// The quality value of this content type.
+    ///
+    /// Used to describe the order of priority in a comma-separated
+    /// list of values.
+    ///
+    /// Content types with a higher priority should be preferred by the server
+    /// when deciding which content type to use in the response.
+    ///
+    /// Also called the "q-factor" or "q-value".
+    public var quality: QualityValue
+
+    /// Creates a new content type from the provided parameters.
+    /// - Parameters:
+    ///   - value: The value representing the content type.
+    ///   - quality: The quality of the content type, between 0.0 and 1.0.
+    /// - Precondition: Priority must be in the range 0.0 and 1.0 inclusive.
```

Comment:
```suggestion
    /// - Precondition: Quality must be in the range 0.0 and 1.0 inclusive.
```

Reply:
Good catch, thanks. I started calling them "priority" originally but then (clearly imperfectly) updated the codebase.